### PR TITLE
Add mypy_path config option (#2323)

### PR DESCRIPTION
**Pull request by megapctr**
I tried to write a simple test for the new `build_lib_path` function, but pytest wouldn't pick it up. Help?

### CODE REVIEW

**Comment by ddfisher** [comment]
I don't see the test you mentioned in the PR, so it's hard for me to say what might've gone wrong.  If you add it I can take a look.

**Comment by megapctr** [comment]
Updated - is this ok? As for the test, I figured it will make more sense to add a test case to `test-data/unit` dir, so I'll try that.

File: mypy/main.py [hunk 1]
```diff
@@ -451,6 +451,7 @@ def get_init_file(dir: str) -> Optional[str]:
     'python_version': lambda s: tuple(map(int, s.split('.'))),
     'strict_optional_whitelist': lambda s: s.split(),
     'custom_typing_module': str,
+    'mypy_path': lambda s: [p.strip() for p in s.split(',')],
```
Comment (megapctr) [inline mypy/main.py]: I think it's better to stick to [a common convention](https://github.com/search?utf8=%E2%9C%93&q=exclude+node_modules+extension%3A.cfg&type=Code&ref=searchresults) of handling lists in config files. Parsing that setting this way would probably cause confusion.

Comment (megapctr) [inline mypy/main.py]: Sure. At that point, this function could have a test too. How can I add a small unit test? All I can find is big test suites or use-case scenarios, and that's too much to test a pure function 🙂

Comment (ddfisher) [inline mypy/main.py]: Splitting on commas and colons seems reasonable, but we shouldn't split on spaces: it's not uncommon to have spaces in directory names.  It'd be nice to mention allowable path separators in the documentation for this config option.

@gvanrossum do we currently have tests for the config file somewhere?

**Comment by gvanrossum** [comment]
OK, as long as leading/trailing spaces are stripped (I think I want to be
able to write "path = x, y, z").

There are some tests for config files in cmdline.test.

**Comment by gvanrossum** [comment]
If you rebase now the tests should pass.

**Comment by megapctr** [comment]
I'm up to date

**Comment by gvanrossum** [comment]
OK, I'm going to look into why the tests fail then. Because that failure seems due to a missed typeshed sync to me (you didn't touch reports.py, there was a recent change to reports.py that depended on some typeshed changes).

**Comment by ddfisher** [comment]
Yeah, it looks like typeshed is out of date.  You can fix that with `git checkout master -- typeshed && git commit -a`.

**Comment by ddfisher** [comment]
This looks good!  We should probably change `os.pathsep` back to colon, actually -- otherwise mypy.ini files won't work cross-platform properly. :/  Sorry for flip-flopping about that!

**Comment by megapctr** [comment]
Ok, will do - thanks for noticing that issue @ddfisher!

**Comment by megapctr** [comment]
I did `git checkout upstream/master typeshed` and the submodule commit changed, so it was probably an error on my side. Sorry for that! 🙇

**Comment by ddfisher** [comment]
Thanks!

**Pull request merged**